src/hist/plot.py

```diff
@@ -210,159 +220,334 @@ def plot2d_full(
     return main_art, top_art, side_art
 
 
-def plot_pull(
-    self: hist.BaseHist,
-    func: Union[Callable[[np.ndarray], np.ndarray], str],
-    likelihood: bool = False,
-    *,
-    ax_dict: "Optional[Dict[str, matplotlib.axes.Axes]]" = None,
-    fit_fmt: Optional[str] = None,
-    **kwargs: Any,
-) -> "Tuple[matplotlib.axes.Axes, matplotlib.axes.Axes]":
-    r"""
-    Plot_pull method for BaseHist object.
-
-    fit_fmt can be a string such as r"{name} = {value:.3g} $\pm$ {error:.3g}"
+# TODO: Move interval estimation functions into another module?
+# Taken from coffea for now
+# https://github.com/CoffeaTeam/coffea/blob/12228d5564963234434d5c881efa761e088c298a/coffea/hist/plot.py#L57
+# TODO: Check licensing
+def poisson_interval(
+    values: np.ndarray, variances: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    """Frequentist coverage interval for Poisson-distributed observations
+    Parameters
+    ----------
+        values : np.ndarray
+            Sum of weights vector
+        variances : np.ndarray
+            Sum weights squared vector
+        coverage : float, optional
+            Central coverage interval, defaults to 68%
+    Calculates the so-called 'Garwood' interval,
+    c.f. https://www.ine.pt/revstat/pdf/rs120203.pdf or
+    http://ms.mcmaster.ca/peter/s743/poissonalpha.html
+    For weighted data, this approximates the observed count by ``values**2/variances``, which
+    effectively scales the unweighted poisson interval by the average weight.
+    This may not be the optimal solution: see https://arxiv.org/pdf/1309.1287.pdf for a proper treatment.
+    When a bin is zero, the scale of the nearest nonzero bin is substituted to scale the nominal upper bound.
+    If all bins zero, a warning is generated and interval is set to ``values``.
```

Comment:
This warning when everything is zero has been criticized by coffea users, as its a bit unwelcome and frequent (since empty histograms happen often).

Reply:
@nsmith- yeah, that's why I later down have it called under a `with np.errstate(divide="ignore"):` context.